signxml/__init__.py

```diff
@@ -588,11 +612,14 @@ def verify(self, require_x509=True, x509_cert=None, ca_pem_file=None, ca_path=No
             root = fromstring(self.data, parser=parser)
         else:
             root = self.data
+        c14n_root = fromstring(etree.tostring(root))
```

Comment:
Since we may need to apply transformation modifying the XML tree during the canonicalization process we need to make a copy of it to prevent modifying the original tree.

Originally I used copy.deepcopy but with lxml it failed at preserving the namespaces on the root node which broke some cases, thus I instead copy the tree in this way.